wsmaster/che-core-api-devfile/src/main/resources/schema/devfile.json

```diff
@@ -95,110 +92,371 @@
+                        "parentSelector": {
+                          "$ref": "#/definitions/selector",
+                          "description": "The selector on labels of the top level objects in the referenced list in which to search for containers. If not specified, the objects to search through can have any labels."
+                        },
+                        "command": {
+                          "type": "array",
+                          "items": {
+                            "type": "string"
+                          },
+                          "default": null,
+                          "description": "The command to run in the component instead of the default one provided in the image of the container. Defaults to null, meaning use whatever is defined in the image.",
+                          "examples": [
+                            "['/bin/sh', '-c']"
+                          ]
+                        },
+                        "args": {
+                          "type": "array",
+                          "items": {
+                            "type": "string"
+                          },
+                          "default": null,
+                          "description": "The arguments to supply to the command running the component. The arguments are supplied either to the default command provided in the image of the container or to the overridden command. Defaults to null, meaning use whatever is defined in the image.",
+                          "examples": [
+                            "['-R', '-f']"
+                          ]
+                        }
+                      }
+                    }
+                  }
+                },
+                "additionalProperties": false,
+                "anyOf": [
+                  {
+                    "required": [
+                      "reference"
+                    ],
+                    "additionalProperties": true
+                  },
+                  {
+                    "required": [
+                      "referenceContent"
+                    ],
+                    "additionalProperties": true
+                  }
                 ]
-              },
-              "alias": {},
-              "image": {},
-              "memoryLimit": {},
-              "mountSources": {},
-              "volumes": {},
-              "env": {},
-              "endpoints": {},
-              "command": {},
-              "args": {}
+              }
             },
-            "required": [
-              "image",
-              "memoryLimit"
-            ],
-            "additionalProperties": false
-          }
-        ],
+            {
+              "if": {
+                "properties": {
+                  "type": {
+                    "enum": [
+                      "dockerimage"
+                    ]
+                  }
+                }
+              },
+              "then": {
+                "required": [
```

Comment:
Please make sure that we have one order style, like `type`, `required`, `additionalProperties`, `properties`. At least in part that you changed